### PR TITLE
Update service mode check and set probe on integration tests

### PR DESCRIPTION
**Pull request by randeepsharma**
# Description
Currently mode is either controller or node in controller & node pods respectively.
Earlier client is created even when the noProbeOnStart was set to true.
Now with the new changes when noProbeOnStart is set to true, no client is created.
Upon setting the noProbeOnStart is set to false, it looks for the mode and creates the client which is used to run operations on array.
Since the mode is blank it fails with the error, "service mode not set".
Made the current changes, when mode is not set, it runs both the controller and node probes. 
With the above change, the csi driver for powerscale runs even when mode is not set.

This fixes the integration tests as well.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
https://github.com/dell/csm/issues/128

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Cert-csi and integration tests are running successfully.
- [x] Tested manually as well.

### CODE REVIEW

**Review by prablr79 — APPROVED** [review]

File: service/service.go [hunk 1]
```diff
@@ -319,6 +319,15 @@ func (s *service) probe(ctx context.Context, clusterConfig *IsilonClusterConfig)
 		if err := s.nodeProbe(ctx, clusterConfig); err != nil {
 			return err
 		}
+	} else if strings.EqualFold(s.mode, "") {
+		log.Warn("Service mode not set, attempting both controller and node probe")
+		controllerErr := s.controllerProbe(ctx, clusterConfig)
```
Comment (bpjain2004) [inline service/service.go:324]: fail if there is a controllerErr; no need to do nodeProbe()

File: service/service.go [hunk 2]
```diff
@@ -319,6 +319,15 @@ func (s *service) probe(ctx context.Context, clusterConfig *IsilonClusterConfig)
...
+		nodeProbeErr := s.nodeProbe(ctx, clusterConfig)
+
+		if controllerErr != nil || nodeProbeErr != nil {
+			return fmt.Errorf("probe failed")
+		}
+		return nil
 	} else {
 		return status.Error(codes.FailedPrecondition,
 			"Service mode not set")
```
Comment (bpjain2004) [inline service/service.go:333]: Change this error message to "invalid mode"

File: service/service.go (hunk 1 above)
Comment (Sakshi-dell) [reply service/service.go:324]: Addressed

File: service/service.go [hunk 3]
```diff
@@ -322,15 +322,18 @@ func (s *service) probe(ctx context.Context, clusterConfig *IsilonClusterConfig)
...
+		if controllerErr != nil {
+			return fmt.Errorf("probe failed")
+		}
 
-		if controllerErr != nil || nodeProbeErr != nil {
+		nodeProbeErr := s.nodeProbe(ctx, clusterConfig)
+		if nodeProbeErr != nil {
 			return fmt.Errorf("probe failed")
 		}
 		return nil
```
Comment (bpjain2004) [inline service/service.go:333]: remove this return statement which is not required

**Review by bpjain2004 — DISMISSED** [review]

**Review by bpjain2004 — APPROVED** [review]

**Review by prablr79 — APPROVED** [review]

**Pull request merged**